src/modules/physics/physics_jolt.c

```diff
@@ -0,0 +1,1474 @@
+  };
+  array_to_rmatrix_struct(transformArray, &transform);
+  JPH_AllHit_CastShapeCollector_Reset(cast_shape_collector);
+  JPH_NarrowPhaseQuery_CastShape(query, queryBox, &transform, &direction, &base_offset, cast_shape_collector);
+  size_t hit_count;
+  JPH_ShapeCastResult* hit_array = JPH_AllHit_CastShapeCollector_GetHits(cast_shape_collector, &hit_count);
+  for (int i = 0; i < hit_count; i++) {
+    JPH_BodyID id = JPH_AllHit_CastShapeCollector_GetBodyID2(cast_shape_collector, i);
+    Collider* collider = (Collider*) JPH_BodyInterface_GetUserData(
+      world->body_interface,
+      id);
+    size_t count;
+    Shape** shape = lovrColliderGetShapes(collider, &count);
+    bool shouldStop = callback(shape[0], userdata);
+    if (shouldStop)
+      break;
+  }
+  return hit_count > 0;
+}
+
+bool lovrWorldQueryBox(World* world, float position[3], float size[3], QueryCallback callback, void* userdata) {
+  return lovrWorldQueryShape(world, queryBox, position, size, callback, userdata);
+}
+
+bool lovrWorldQuerySphere(World* world, float position[3], float radius, QueryCallback callback, void* userdata) {
+  float scale[3] = {radius, radius, radius};
+  return lovrWorldQueryShape(world, querySphere, position, scale, callback, userdata);
+}
+
+Collider* lovrWorldGetFirstCollider(World* world) {
+  return world->head;
+}
+
+void lovrWorldGetGravity(World* world, float* x, float* y, float* z) {
+  JPH_Vec3 gravity;
+  JPH_PhysicsSystem_GetGravity(world->physics_system, &gravity);
+  *x = gravity.x;
+  *y = gravity.y;
+  *z = gravity.z;
+}
+
+void lovrWorldSetGravity(World* world, float x, float y, float z) {
+  const JPH_Vec3 gravity = {
+    .x = x,
+    .y = y,
+    .z = z
+  };
+  JPH_PhysicsSystem_SetGravity(world->physics_system, &gravity);
+}
+
+float lovrWorldGetResponseTime(World* world) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support global ResponseTime option");
+}
+
+void lovrWorldSetResponseTime(World* world, float responseTime) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support global ResponseTime option");
+}
+
+float lovrWorldGetTightness(World* world) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support Tightness option");
+}
+
+void lovrWorldSetTightness(World* world, float tightness) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support Tightness option");
+}
+
+void lovrWorldGetLinearDamping(World* world, float* damping, float* threshold) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support global LinearDamping option");
+}
+
+void lovrWorldSetLinearDamping(World* world, float damping, float threshold) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support global LinearDamping option");
+}
+
+void lovrWorldGetAngularDamping(World* world, float* damping, float* threshold) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support global AngularDamping option");
+}
+
+void lovrWorldSetAngularDamping(World* world, float damping, float threshold) {
+  lovrLog(LOG_WARN, "PHY", "Jolt doesn't support global AngularDamping option");
```

Comment:
I think the World damping accessors in ODE just set the default damping for new bodies, so maybe these could be supported (but we could also think about removing them).

Reply:
For now I've added both dampings, as well as sleeping allowed.

Small note - the docs could be clarified. From wording its easy to assume that setting the world damping at any time will change the damping of previously created colliders.

Reply:
Clarified the docs for this